node/src/actors/sessions_manager/actor.rs

```diff
@@ -25,7 +25,7 @@ impl Actor for SessionsManager {
 
                 // Set server address, connections limits and handshake timeout
                 act.sessions
-                    .set_server_address(config.connections.server_addr);
+                    .set_server_address(config.connections.public_addr);
```

Comment:
Maybe we should rename this method and `self.server_address` into `self.public_address`?

```suggestion
                    .set_public_address(config.connections.public_addr);
```

Reply:
Agreed.